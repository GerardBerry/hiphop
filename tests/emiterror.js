"use hopscript"

const hh = require("hiphop");

const prg = <hh.module>
  <hh.outputsignal name="O" valued/>
  <hh.loop>
    <hh.sequence>
      <hh.emit signal_name="O" arg=5 />
      <hh.emit signal_name="O" arg=5 />
      <hh.pause/>
    </hh.sequence>
  </hh.loop>
</hh.module>;

const machine = new hh.ReactiveMachine(prg, "emiterror");

try {
   machine.react();
} catch (e) {
   console.log(e.message);
}
