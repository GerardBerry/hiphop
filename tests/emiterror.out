Can't set single signal `O' value more that once.
